internal/test/client.go

```diff
@@ -0,0 +1,20 @@
+package test
+
+import (
+	"github.com/ninech/nctl/api"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func SetupClient(initObjs ...client.Object) (*api.Client, error) {
+	scheme, err := api.NewScheme()
+	if err != nil {
+		return nil, err
+	}
+
+	client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(initObjs...).Build()
+
+	return &api.Client{
+		WithWatch: client, Namespace: "default",
```

Comment:
I like the idea to have a common logic for test setup!

one suggestion: maybe we should expose the namespace argument as well? This will make it more universal. Additionally, having this hardcoded we need to remember to use the "default" namespace inside the test - it's a bit error prone.

I will also change my PR to match this new SetupClient() func.

Reply:
It's already exposed as it's an exported field on the client ;) I kind of don't want to set it all the time in an argument when it's going to be `default` in almost all cases.